src/_pytest/assertion/util.py

```diff
@@ -160,8 +160,8 @@ def assertrepr_compare(config, op: str, left: Any, right: Any) -> Optional[List[
     """Return specialised explanations for some operators/operands."""
     verbose = config.getoption("verbose")
     if verbose > 1:
-        left_repr = safeformat(left)
-        right_repr = safeformat(right)
+        left_repr = saferepr_unlimited(left)
```

Comment:
Indeed using `safeformat` for the left and right operands when `verbose > 1` is not what is expected, because `safeformat` introduces newlines, and the result of `assertrepr_compare` is expected to be a list of lines; the string produced by `safeformat` (containing `\n`) is put as *a single string* in the result, which causes them to be escaped by the terminal later (I assume).

So the fix is entirely correct, we use `repr` to produce the representation without `\n`, which is used to build the `summary` variable which is expected to be a single line.